src/main/java/org/opensearch/neuralsearch/processor/DocumentChunkingProcessor.java

```diff
@@ -0,0 +1,333 @@
+        List<String> result = chunker.chunk(content, chunkerParameters);
+        current_chunk_count += result.size();
+        if (max_chunk_limit != DEFAULT_MAX_CHUNK_LIMIT && current_chunk_count > max_chunk_limit) {
+            throw new IllegalArgumentException(
+                "Unable to create the processor as the number of chunks ["
+                    + current_chunk_count
+                    + "] exceeds the maximum chunk limit ["
+                    + max_chunk_limit
+                    + "]"
+            );
+        }
+        return result;
+    }
+
+    private List<String> chunkList(List<String> contentList) {
+        // flatten the List<List<String>> output to List<String>
+        List<String> result = new ArrayList<>();
+        for (String content : contentList) {
+            result.addAll(chunkString(content));
+        }
+        return result;
+    }
+
+    @SuppressWarnings("unchecked")
+    private List<String> chunkLeafType(Object value) {
+        // leaf type is either String or List<String>
+        List<String> chunkedResult = null;
+        if (value instanceof String) {
+            chunkedResult = chunkString(String.valueOf(value));
+        } else if (isListString(value)) {
+            chunkedResult = chunkList((List<String>) value);
+        }
+        return chunkedResult;
+    }
+
+    /**
+     * This method will be invoked by PipelineService to perform chunking and then write back chunking results to the document.
+     * @param ingestDocument {@link IngestDocument} which is the document passed to processor.
+     */
+    @Override
+    public IngestDocument execute(IngestDocument ingestDocument) {
+        validateFieldsValue(ingestDocument);
+        current_chunk_count = 0;
+        if (Objects.equals(chunkerType, FIXED_LENGTH_ALGORITHM)) {
+            // add maxTokenCount setting from index metadata to chunker parameters
+            Map<String, Object> sourceAndMetadataMap = ingestDocument.getSourceAndMetadata();
+            String indexName = sourceAndMetadataMap.get(IndexFieldMapper.NAME).toString();
+            int maxTokenCount = IndexSettings.MAX_TOKEN_COUNT_SETTING.get(environment.settings());
+            IndexMetadata indexMetadata = clusterService.state().metadata().index(indexName);
+            if (indexMetadata != null) {
+                // if the index exists, read maxTokenCount from the index setting
+                IndexService indexService = indicesService.indexServiceSafe(indexMetadata.getIndex());
+                maxTokenCount = indexService.getIndexSettings().getMaxTokenCount();
+            }
+            chunkerParameters.put(FixedTokenLengthChunker.MAX_TOKEN_COUNT_FIELD, maxTokenCount);
+        }
+
+        Map<String, Object> sourceAndMetadataMap = ingestDocument.getSourceAndMetadata();
+        chunkMapType(sourceAndMetadataMap, fieldMap);
+        sourceAndMetadataMap.forEach(ingestDocument::setFieldValue);
```

Comment:
what does this line do? It seems results are already put into the sourceAndMetadataMap in line 281.

Reply:
I will check whether this change apply to the ingestdocument

Reply:
Already changed